pallets/capacity/src/tests/rewards_provider_tests.rs

```diff
@@ -72,3 +41,27 @@ fn test_staking_reward_total_era_out_of_range() {
 		assert_err!(Capacity::staking_reward_total(1, 5u32, 13u32), Error::<Test>::EraOutOfRange);
 	});
 }
+
+// This tests Capacity implementation of the trait, but uses the mock's constants,
+// to ensure that it's correctly specified in the pallet.
+#[test]
+fn staking_reward_for_era_implementation() {
+	struct TestCase {
+		total_staked: u64,    // by all boosters
+		amount_staked: u64,   // by a single booster
+		expected_reward: u64, // for a single era
+	}
+	let test_cases: Vec<TestCase> = vec![
+		TestCase { total_staked: 1_000_000, amount_staked: 0, expected_reward: 0 }, // shouldn't happen, but JIC
+		TestCase { total_staked: 1_000_000, amount_staked: 30, expected_reward: 0 }, // rounded result
+		TestCase { total_staked: 1_000_000, amount_staked: 150, expected_reward: 1 }, // it rounds the result
+		TestCase { total_staked: 1_000_000, amount_staked: 1000, expected_reward: 4 }, // hits the cap starting with this example
+		TestCase { total_staked: 1_000_000, amount_staked: 11000, expected_reward: 42 }, // > 0.0038% of total, reward = 11000*.0038
```

Comment:
I guess this got moved, but the main change to this test was to add the last test case.